api/src/services/meta.ts

```diff
@@ -58,27 +66,34 @@ export class MetaService {
 	async filterCount(collection: string, query: Query): Promise<number> {
 		const dbQuery = this.knex(collection);
 
-		let filter = query.filter || {};
+		const filter = query.filter || {};
 		let hasJoins = false;
 
-		if (this.accountability?.admin !== true) {
-			const permissionsRecord = this.accountability?.permissions?.find((permission) => {
-				return permission.action === 'read' && permission.collection === collection;
-			});
-
-			if (!permissionsRecord) throw new ForbiddenError();
-
-			const permissions = permissionsRecord.permissions ?? {};
-
-			if (Object.keys(filter).length > 0) {
-				filter = { _and: [permissions, filter] };
-			} else {
-				filter = permissions;
-			}
+		if (this.accountability) {
+			await validateAccess(
+				{
+					accountability: this.accountability,
+					action: 'read',
+					collection,
+				},
+				{
+					schema: this.schema,
+					knex: this.knex,
+				},
+			);
+
+			// TODO rely on applyCases
+			// const permissions = permissionsRecord.permissions ?? {};
+
+			// if (Object.keys(filter).length > 0) {
+			// 	filter = { _and: [permissions, filter] };
+			// } else {
+			// 	filter = permissions;
+			// }
```

Comment:
When we bring this back, we need to make sure to include `&& this.accountability.admin !== true` in the if above, otherwise all fields will be filtered out.